src/adapters/node/adapter.ts

```diff
@@ -0,0 +1,65 @@
+import type {IncomingMessage, ServerResponse} from 'http';
+
+import fetch from 'node-fetch';
+
+import {
+  canonicalizeHeaders,
+  flatHeaders,
+  NormalizedRequest,
+  NormalizedResponse,
+} from '../../runtime/http';
+
+export async function nodeConvertRequest(
+  req: IncomingMessage,
+): Promise<NormalizedRequest> {
+  const body = await new Promise<string>((resolve, reject) => {
+    let str = '';
+    req.on('data', (chunk) => {
+      str += chunk.toString();
+    });
+    req.on('error', (error) => reject(error));
+    req.on('end', () => resolve(str));
+  });
+  return {
+    headers: canonicalizeHeaders({...req.headers} as any),
+    method: req.method ?? 'GET',
+    url: req.url!,
+    body,
+  };
+}
+
+export async function nodeConvertResponse(
+  response: NormalizedResponse,
+  res: ServerResponse,
+): Promise<void> {
+  res.statusCode = response.statusCode;
+  res.statusMessage = response.statusText;
+
+  if (response.headers) {
+    Object.entries(response.headers).forEach(([header, value]) =>
+      res.setHeader(header, value),
+    );
+  }
+
+  if (response.body) {
+    res.write(response.body);
+  }
+
+  res.end();
```

Comment:
That's a good question. I think we should approach this from the app's perspective. If a lib request handles both the request and response (e.g. webhook processing):

```js
app.post('/webhooks', async (req, res) => {
  webhooks.process(req, res);
})
```

versus

```js
app.post('/webhooks', async (req, res) => {
  webhooks.process(req, res);
  // Do more things with the response here? Would that work if we've written headers and body at this point?
  res.end();
})
```

Reply:
In the example you gave, wouldn't the `res.end` be within the `webhooks.process` method, e.g.,

```javascript
function process(req, res) {
  // do stuff
  nodeConvertResponse(res);
  // do more stuff
  res.end();
}

In a sense, it doesn't really matter since it's used internally, just wondering if the name should be clearer so library devs know that actually sends the response as well, not just convert it.
```

Reply:
Oh that's fair, I think we should rename it then!